extensions/stackdriver/stackdriver.cc

```diff
@@ -54,8 +56,20 @@ void StackdriverRootContext::onConfigure(
     return;
   }
 
-  // Only register exporter once in main thread when initiating base WASM
-  // module.
+  // Get node metadata. GetMetadataStruct always returns the whole node metadata
+  // even with a key passed in.
+  // TODO: change to GetMetadataStruct after fixing upstream API to respect node metadata key.
+  auto node_metadata = getMetadataValue(Common::Wasm::MetadataType::Node, kIstioMetadataKey);
+  status = ExtractNodeMetadata(node_metadata.struct_value(),
+                                    &local_node_info_);
+  if (status != Status::OK) {
+    logWarn("cannot parse local node metadata " +
+            node_metadata.DebugString() + ": " + status.ToString());
```

Comment:
Not sure if you need an early return

Reply:
I thought about it too and I make it not to early return. Exporter will still be registered and user will get better error message from Opencensus library on how wrong node metadata would affect metric exporting.